tests/tracking/test_model_registry.py

```diff
@@ -2,101 +2,77 @@
 
-# pylint: disable=unused-argument
-
-# Root directory for all stores (backend or artifact stores) created during this suite
-SUITE_ROOT_DIR = tempfile.mkdtemp("test_rest_tracking")
-# Root directory for all artifact stores created during this suite
-SUITE_ARTIFACT_ROOT_DIR = tempfile.mkdtemp(suffix="artifacts", dir=SUITE_ROOT_DIR)
 
 _logger = logging.getLogger(__name__)
 
 
-def _get_sqlite_uri():
-    path = path_to_local_file_uri(os.path.join(SUITE_ROOT_DIR, "test-database.bd"))
-    path = path[len("file://") :]
-
-    # NB: It looks like windows and posix have different requirements on number of slashes for
-    # whatever reason. Windows needs uri like 'sqlite:///C:/path/to/my/file' whereas posix expects
-    # sqlite://///path/to/my/file
-    prefix = "sqlite://" if sys.platform == "win32" else "sqlite:////"
-    return prefix + path
-
-
-# Backend store URIs to test against
-BACKEND_URIS = [
-    _get_sqlite_uri(),  # SqlAlchemy
-]
-
-# Map of backend URI to tuple (server URL, Process). We populate this map by constructing
-# a server per backend URI
-BACKEND_URI_TO_SERVER_URL_AND_PROC = {
-    uri: _init_server(backend_uri=uri, root_artifact_uri=SUITE_ARTIFACT_ROOT_DIR)
-    for uri in BACKEND_URIS
-}
+@pytest.fixture(scope="module")
+def module_scoped_tmp_dir():
+    with tempfile.TemporaryDirectory() as tmpdir:
+        yield Path(tmpdir)
 
 
-def pytest_generate_tests(metafunc):
-    """
-    Automatically parametrize each each fixture/test that depends on `backend_store_uri` with the
-    list of backend store URIs.
-    """
-    if "backend_store_uri" in metafunc.fixturenames:
-        metafunc.parametrize("backend_store_uri", BACKEND_URIS)
+@pytest.fixture(scope="module", params=["file", "sqlalchemy"])
+def backend_uri(request, module_scoped_tmp_dir):
+    if request.param == "file":
+        return path_to_local_file_uri(str(module_scoped_tmp_dir.joinpath("file")))
+    else:
+        path = path_to_local_file_uri(str(module_scoped_tmp_dir.joinpath("sqlalchemy.db")))
+        return ("sqlite://" if sys.platform == "win32" else "sqlite:////") + path[len("file://") :]
 
 
-@pytest.fixture(scope="module", autouse=True)
-def server_urls():
-    """
-    Clean up all servers created for testing in `pytest_generate_tests`
-    """
-    yield
-    for server_url, process in BACKEND_URI_TO_SERVER_URL_AND_PROC.values():
-        _logger.info(f"Terminating server at {server_url}...")
-        _logger.info(f"type = {type(process)}")
-        process.terminate()
-        _await_server_down_or_die(process)
-    shutil.rmtree(SUITE_ROOT_DIR)
+@pytest.fixture(autouse=True)
+def reset_backend(backend_uri, module_scoped_tmp_dir):
+    for child in module_scoped_tmp_dir.iterdir():
+        if child.is_dir():
+            shutil.rmtree(child)
+        else:
+            child.unlink()
 
+    if backend_uri.startswith("file"):
+        FileStore(backend_uri)
+    else:
+        SqlAlchemyStore(backend_uri)
```

Comment:
To avoid launching a new server for each test, reinitializing the backend store before each test invocation.